src/LayoutManager/LayoutManager.ts

```diff
@@ -0,0 +1,277 @@
+            .subtract(nextCenter)
+            .add(correction)
+            .transform(
+              // in `initialization` we do not account for target's transformation matrix
+              context.type === 'initialization'
+                ? iMatrix
+                : invertTransform(target.calcOwnMatrix()),
+              true
+            )
+            .add(relativeCorrection);
+    return {
+      result,
+      prevCenter,
+      nextCenter,
+      offset,
+    };
+  }
+
+  protected commitLayout(
+    context: StrictLayoutContext,
+    layoutResult: Required<LayoutResult>
+  ) {
+    const { target } = context;
+    const {
+      result: { width, height },
+      prevCenter,
+      nextCenter,
+    } = layoutResult;
+    // set dimensions
+    target.set({ width, height });
+    // layout descendants
+    this.layoutObjects(context, layoutResult);
+    //  set position
+    // in `initialization` we do not account for target's transformation matrix
+    if (context.type === 'initialization') {
+      const origin = nextCenter.add(
+        new Point(width, height).multiply(
+          new Point(
+            resolveOrigin(target.originX),
+            resolveOrigin(target.originY)
+          )
+        )
+      );
+      target.set({ left: context.x ?? origin.x, top: context.y ?? origin.y });
+    } else if (!nextCenter.eq(prevCenter)) {
+      target.setPositionByOrigin(nextCenter, CENTER, CENTER);
+    }
+    // invalidate
+    context.type !== 'initialization' && target.setCoords();
+    target._set('dirty', true);
+  }
+
+  protected layoutObjects(
+    context: StrictLayoutContext,
+    layoutResult: Required<LayoutResult>
+  ) {
+    const { target } = context;
+    //  adjust objects to account for new center
+    (context.type !== 'initialization' || !context.objectsRelativeToGroup) &&
+      target.forEachObject((object) => {
+        object.group === target &&
+          this.layoutObject(context, layoutResult, object);
+      });
+    // adjust clip path to account for new center
+    context.strategy.shouldLayoutClipPath(context) &&
+      target.clipPath &&
+      !target.clipPath.absolutePositioned &&
+      this.layoutObject(context, layoutResult, target.clipPath as FabricObject);
+  }
+
+  /**
+   * @param {FabricObject} object
+   * @param {Point} offset
+   */
+  protected layoutObject(
+    context: StrictLayoutContext,
+    { offset }: Required<LayoutResult>,
+    object: FabricObject
+  ) {
+    object.setRelativeXY(object.getRelativeXY().add(offset));
+  }
+
+  protected onAfterLayout(
+    context: StrictLayoutContext,
+    layoutResult?: LayoutResult
+  ) {
+    const { target } = context;
+
+    //  fire layout hook and event (event will fire only for layouts after initialization layout)
+    target.onAfterLayout({
+      context,
+      result: layoutResult,
+    });
+    target.fire('layout', {
+      context,
+      result: layoutResult,
+    });
+
+    //  bubble
+    if (target.group?.layoutManager) {
```

Comment:
I just had a case today where I traced a bug back to this. When applying syncing xy changes from the server to a group and its children, which are also groups, I don't want a `child.triggerLayout()` to bubble up. I've not finished yet updating all the objects with the final relative xy, so bubbling up will use a mix of old and new relative xy. I will manually call `parent.triggerLayout()`.

So in brief: there should be a flag in the `context` to stop the bubbling

Reply:
I think I understand the need but not sure how you are using `triggerLayout` in this use case.
Could you show me?
I do understand that you want to stop bubbling for perf reasons - so the top group will be layed out only once instead of each time a child lays out.

Reply:
Do you think we should add a method that triggers a deep layout on all its childen?
Then you call `triggerLayout({deep:true})` and you are done.

Reply:
> I do understand that you want to stop bubbling for perf reasons

It's not about perf, rather it's a bug to bubble up the layout invalidation when in the middle of updating several children since you need to wait for *all* of them to be updated first, because the xy values are relative to the parent center.
Use case: syncing changing strokeWidth of subselected group (shape in group) => results in syncing the new xy values of the whole group and children because the parent has changed size and center. The change of strokeWidth was significant.
- For each child, we set the new xy values and for one of them the new strokeWidth. We also call `child.triggerLayout()` 
- If the layout invalidation is bubbled up, the parent will use incorrect values. Some children are sill with the new properties, some are with new ones. This results in slightly wrong positions, you may even not notice them at first. I can show you in our next call.

> Do you think we should add a method that triggers a deep layout on all its childen?
Then you call triggerLayout({deep:true}) and you are done.

Not sure, I don't think so. A child changing position invalidates the parent layout, but not vice-versa.

Reply:
guys keep it minimal as minimal as possible

Reply:
Minimal: allow to skip bubbling. Don't assume it's always needed.

Reply:
The word that guides me is useful.
I want it to serve me/devs and ease developing.
Of course it should be straight forward as possible not only to use but to write and maintain.
I want to avoid deving stuff that to use in a conventional app I need to break into pieces, override and patch to a point that I almost rewrite it all and whatever fabric put there became useless for me.